homeassistant/components/jellyfin/entity.py

```diff
@@ -32,3 +34,44 @@ def __init__(
             name=self.coordinator.server_name,
             sw_version=self.coordinator.server_version,
         )
+
+
+class JellyfinClientEntity(JellyfinEntity):
+    """Defines a base client Jellyfin entity."""
+
+    session_data: dict[str, Any] | None
+    session_id: str
+
+    def __init__(
+        self,
+        coordinator: JellyfinDataUpdateCoordinator[JellyfinDataT],
+        description: EntityDescription,
+        session_id: str,
+        session_data: dict[str, Any],
+    ) -> None:
+        """Initialize the Jellyfin entity."""
+        super().__init__(coordinator, description)
+
+        self.session_id = session_id
+        self.session_data: dict[str, Any] = session_data
+        self.device_id: str = session_data["DeviceId"]
+        self.device_name: str = session_data["DeviceName"]
+        self.client_name: str = session_data["Client"]
+        self.app_version: str = session_data["ApplicationVersion"]
+
+        self.capabilities: dict[str, Any] = session_data["Capabilities"]
+
+        if self.capabilities.get("SupportsPersistentIdentifier", False):
```

Comment:
When doesn't it support this identifier?

Reply:
Capabilities are reported by clients, some clients can be ephemeral without a persistent identifier (they don't send a device-id back when talking with the Jellyfin server). I don't know if there's a list anywhere of which clients do or don't support a persistent identifier.